src/lib/Form/Type/Content/BaseContentType.php

```diff
@@ -61,8 +67,27 @@ public function buildView(FormView $view, FormInterface $form, array $options)
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver
-            ->setDefaults(['translation_domain' => 'ibexa_content_forms_content'])
-            ->setRequired(['languageCode', 'mainLanguageCode']);
+            ->setDefaults([
+                'translation_domain' => 'ibexa_content_forms_content',
+                'struct' => null,
+            ])
+            ->setAllowedTypes(
+                'struct',
+                [
+                    'null',
+                    ContentCreateStruct::class,
+                    ContentUpdateStruct::class,
+                    UserCreateStruct::class,
+                    UserUpdateStruct::class,
+                ],
+            )
+            ->setRequired(['languageCode', 'mainLanguageCode', 'struct'])
+            ->setNormalizer('struct', static function (Options $options, $value) {
+                return $options['userUpdateStruct']
+                    ?? $options['userCreateStruct']
+                        ?? $options['contentUpdateStruct']
+                            ?? $options['contentCreateStruct'];
+            });
```

Comment:
To ensure we don't perform overwrite (`$value` different than `null`) and we only expect one of the other, deprecated options to be set:
```suggestion
            ->setNormalizer('struct', static function (Options $options, $value) {
                if ($value !== null) {
                    return $value;
                }

                if (isset($options['userUpdateStruct'])
                    xor isset($options['userCreateStruct)
                    xor isset($options['contentUpdateStruct'])
                    xor isset($options['contentCreateStruct'])
                ) {
                    return $options['userUpdateStruct']
                        ?? $options['userCreateStruct']
                        ?? $options['contentUpdateStruct']
                        ?? $options['contentCreateStruct']
                    ;
                }

                return $value;
            });
```